File: python/same-binary-tree.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class Tree:
    def __init__(self, root):
        self.root = root


class Solution:
    def sameTree(self, root1, root2):
        if root1 is None and root2 is None:
            return True
        if root1 is not None and root2 is not None:
            return root1.val == root2.val and self.sameTree(root1.left, root2.left) and self.sameTree(root1.right, root2.right)
        return False
    

root1 = Node(1)
root1.right = Node(2)
root1.right.left = Node(3)

root2 = Node(1)
root2.right = Node(2)
root2.right.left = Node(3)

print(Solution().sameTree(root1, root2))

File: php/same-binary-tree.php
<?php

class Node {
    public $val;
    public $left;
    public $right;

    public function __construct($val) {
        $this->val = $val;
        $this->left = null;
        $this->right = null;
    }
}

class Tree {
    public $root;
}

class Solution {
    public static function isSameTree($p, $q) {
        if(is_null($p) && is_null($q)) {
            return true;
        }
        if(is_null($p) || is_null($q)) {
            return false;
        }
        if($p->val != $q->val) {
            return false;
        }
        return self::isSameTree($p->left, $q->left) && self::isSameTree($p->right, $q->right);
    }
}

$tree = new Tree();
$tree->root = new Node(1);
$tree->root->left = new Node(2);
$tree->root->right = new Node(3);

$tree2 = new Tree();
$tree2->root = new Node(1);
$tree2->root->left = new Node(2);
$tree2->root->right = new Node(3);

var_dump(Solution::isSameTree($tree->root, $tree2->root));